tests/functionality_test/functionality_testing.py

```diff
@@ -0,0 +1,124 @@
+import json
+import time
+import unittest
+
+from context import Bolt
+
+client = Bolt("<your-api-key-here>", "<your-bolt-id>") # Pass in the API Key and the client ID.
+
+class BoltTests(unittest.TestCase, unittest.TestLoader):
+    """
+    This is a test class to test,
+    the functionality of the bolt python api library.
+    Methods:
+    - setUp()
+    - tearDown()
+    - test_digitalWrite()
+    - test_analogWrite()
+    - test_digitalRead()
+    - test_analogRead()
+    - test_serialBegin()
+    - test_serialWrite()
+    - test_serialRead()
+    - test_Restart()
+    - test_isOnline()
+    - test_isAlive()
+    """
+
+# Setup function for all the testcases.
+    def setUp(self):
+        self.sortTestMethodsUsing = None
+
+# Tear down function for all the testcases
+    def tearDown(self):
+        time.sleep(3)
+
+# Testing the digital write function.
+
+    def test_digitalWrite(self):
+        assert_value = json.loads(client.digitalWrite('4', "HIGH"))
+        self.assertEqual(assert_value["success"], '1')
+        self.assertEqual(assert_value["value"], '1')
+        print("Digital Write Successfull!")
+
+# Testing the analog write function
```

Comment:
@vimalsheoran It would be better if you can write this in the appropriate function's docstring. This would follow the coding guidelines and also be better with the rest of the document as the other functions have docstrings.